src/test/java/uk/gov/pay/api/filter/RateLimiterFilterTest.java

```diff
@@ -2,80 +2,73 @@
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import java.io.PrintWriter;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
 
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.verifyNoMoreInteractions;
 import static org.mockito.Mockito.when;
 
 @RunWith(MockitoJUnitRunner.class)
 public class RateLimiterFilterTest {
 
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+    private String authorization = "Bearer whateverAuthorizationToken";
     private RateLimiterFilter rateLimiterFilter;
-
     private RateLimiter rateLimiter;
-
-    @Mock
-    private HttpServletRequest mockPostRequest;
     @Mock
-    private HttpServletRequest mockGetRequest;
-    @Mock
-    private HttpServletResponse mockResponse;
-    @Mock
-    private FilterChain mockFilterChain;
-
-    String authorization = "Bearer whateverAuthorizationToken";
+    private ContainerRequestContext mockContainerRequestContext;
 
     @Before
     public void setup() {
         rateLimiter = mock(RateLimiter.class);
         rateLimiterFilter = new RateLimiterFilter(rateLimiter, new ObjectMapper());
 
-        when(mockPostRequest.getHeader("Authorization")).thenReturn(authorization);
-        when(mockPostRequest.getMethod()).thenReturn("POST");
+        Account account = new Account("account-id", TokenPaymentType.CARD);
+        SecurityContext mockSecurityContext = mock(SecurityContext.class);
+        when(mockSecurityContext.getUserPrincipal()).thenReturn(account);
 
-        when(mockGetRequest.getHeader("Authorization")).thenReturn(authorization);
-        when(mockGetRequest.getMethod()).thenReturn("GET");
+        when(mockContainerRequestContext.getSecurityContext()).thenReturn(mockSecurityContext);
+
+        when(mockContainerRequestContext.getHeaderString("Authorization")).thenReturn(authorization);
+        when(mockContainerRequestContext.getMethod()).thenReturn("GET");
     }
 
     @Test
-    public void shouldProcessFilterChain_whenRequestsAreWithinTheRate() throws Exception {
-        rateLimiterFilter.doFilter(mockPostRequest, mockResponse, mockFilterChain);
-        rateLimiterFilter.doFilter(mockGetRequest, mockResponse, mockFilterChain);
-
-        verify(mockFilterChain, times(1)).doFilter(mockPostRequest, mockResponse);
-        verify(mockFilterChain, times(1)).doFilter(mockGetRequest, mockResponse);
+    public void shouldCheckRateLimitsWhenFilterIsInvoked() throws Exception {
+        rateLimiterFilter.filter(mockContainerRequestContext);
+        verify(rateLimiter).checkRateOf(any(), any(), any());
```

Comment:
This could be a bit tighter as I think it should be called with `account-id` and the `mockSecurityContext` could be made to return known values for the api key and method that could be checked too? That might help assure that the correct values are being taken from the security context?

Reply:
👍 Updated